contracts/LTOToken.sol

```diff
@@ -28,7 +33,32 @@ contract LTOToken is ERC20, ERC20Detailed, ERC20Burnable, ERC20Pausable {
     require(_intermediate != address(0));
     require(balanceOf(_intermediate) == 0, "Intermediate balance should be 0");
 
+    if (intermediatePending[_intermediate] == PENDING_BRIDGE) {
+      _addIntermediate(_intermediate);
+    } else {
+      intermediatePending[_intermediate] = PENDING_CONFIRM;
+    }
+  }
+
+  function confirmIntermediateAddress() public {
+    require(msg.sender != address(0));
+    require(balanceOf(msg.sender) == 0, "Intermediate balance should be 0");
```

Comment:
Since this is done by wallet holder itself we now can remove this check, and in case there are funds present they get burned automatically upon confirmation.